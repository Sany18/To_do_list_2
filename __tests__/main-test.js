import React from 'react';
import ReactDom from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Main from '../src/main.js';

Enzyme.configure({ adapter: new Adapter() })

describe('Main before sign in', () => {
	let component = mount(<Main />);

	it('should render in "debug" mode', () => {
		expect(component).toBeDefined();
	});

	describe('buttons', () => {
		it('"Sign in" button', () => {
			let but0 = component.find('button').at(0).text()
			expect(but0).toEqual('Sign in');
		});
		it('"Sign up" button', () => {
			let but1 = component.find('button').at(1).text()
			expect(but1).toEqual('Sign up');
		});
	});

	describe('forms', () => {
		it('have 7 inputs', () => {
			expect(component.find('input').length).toEqual(7);
		});
	});
});

describe('Main after sign in', () => {
	localStorage.setItem('access_token', 'imthetoken');

	let component = mount(<Main />);

	const modal = global.document.createElement('div');
	modal.setAttribute("id", 'modal');
	global.document.body.appendChild(modal);

	let btns = component.find('button');

	describe('buttons', () => {
		it('"New task" button', () => {
			expect(btns.at(0).text()).toEqual('New task');
		});

		describe('New task pop-up', () => {
			let modal = '';

			it('"New task" -> click on should be open pop-up', () => {
				component.find('#newTask').at(0).simulate('click');
				modal = document.getElementById('modal');
				expect(modal).toMatchSnapshot()
			});

			it('"Save" button', () => {
				expect(modal.getElementsByTagName('button')[1].innerHTML).toEqual('Save')
			})

			it('"Close" button', () => {
				expect(modal.getElementsByTagName('button')[0].innerHTML).toEqual('Close')		
			})

			it('click on "Close" button should close pop-up', () => {	
				component.find('#newTask').at(0).simulate('click');
				modal.getElementsByTagName('button')[0].click();
				expect(modal.innerHTML.length).toEqual(0);
			})
		})

		it('"Delete selected" button', () => {
			expect(btns.at(1).text()).toEqual('Delete selected');
		});
		it('"Log out" button', () => {
			expect(btns.at(2).text()).toEqual('Log out');
		});
	});

	it('#tasks', () => {
		expect(component.find('#tasks')).toBeDefined();
	});
});