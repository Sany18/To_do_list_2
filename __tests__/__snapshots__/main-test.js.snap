// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main after sign in buttons New task pop-up "New task" -> click on should be open pop-up 1`] = `
<div
  id="modal"
>
  <div>
    <div>
      <label>
        Theme
      </label>
      <input
        name="title"
        value=""
      />
      <label>
        Task
      </label>
      <textarea
        name="theme"
      />
      <label>
        Due date
      </label>
      <input
        min="2018-10-08"
        name="due_date"
        type="date"
        value="2018-10-08"
      />
      <div
        class="btn-group"
      >
        
        <div
          class="btn-group btn-group-sm"
        >
          <button
            class="btn btn-default"
            id="closeTask"
            type="button"
          >
            Close
          </button>
        </div>
        <div
          class="btn-group btn-group-sm"
        >
          <button
            class="btn btn-default"
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
